pkg/targetgroupbinding/networking_manager.go

```diff
@@ -518,19 +520,36 @@ func (m *defaultNetworkingManager) resolveEndpointSGForENI(ctx context.Context,
 		return "", err
 	}
 	clusterResourceTagKey := fmt.Sprintf("kubernetes.io/cluster/%s", m.clusterName)
-	sgIDsWithClusterTag := sets.NewString()
+	sgIDsWithMatchingEndpointSGTags := sets.NewString()
 	for sgID, sgInfo := range sgInfoByID {
+		isMatch := true
 		if _, ok := sgInfo.Tags[clusterResourceTagKey]; ok {
-			sgIDsWithClusterTag.Insert(sgID)
+			for endpointSGTagKey, endpointSGTagValue := range m.endpointSGTags {
+				if sgInfo.Tags[endpointSGTagKey] != endpointSGTagValue {
```

Comment:
Doesn't correctly handle the case of `endpointSGTagValue == ""`. In that case, should fail to match SGs which don't have the tag key present.

Reply:
Ah good catch! I've updated the if statement to also check if the key exists and added [2 test cases](https://github.com/kubernetes-sigs/aws-load-balancer-controller/pull/3147/files#diff-2da03592457251d5626b5e89577539ef29eced083f9a326b5923a91fcea9c11cR1676-R1755) to cover empty tag values.